torch/distributed/pipelining/schedules.py

```diff
@@ -1828,6 +1827,14 @@ class ScheduleInterleaved1F1B(PipelineScheduleMulti):
     state and supports multiple stages per rank. When microbatches are ready for
     multiple local stages, Interleaved 1F1B prioritizes the earlier microbatch
     (also called "depth first").
+
+    This schedule is mostly similar to the original paper.
+    It differs by being relaxing the requirement of num_microbatch % pp_size == 0.
+    Using the flex_pp schedule, we will have num_rounds = max(1, n_microbatches // pp_group_size) and
```

Comment:
nit: we haven't introduced 'flex_pp' as a new schedule.  do we want to define this name, or should we just describe the concept